### DISCUSSION TITLE
JSON Schema support

### DISCUSSION THREAD

**Discussion by sebilasse** [pkosiec/mongo-seeding#158] [Ideas]
Just stumbled about this nice project. How about adding [JSON Schema support](https://docs.mongodb.com/manual/reference/operator/query/jsonSchema/) ?

Schemas could be simply having the same name than a module, like if you have a module called foo.ts and we would find a foo.json in the same folder.
I could do a PR.

**Comment by pkosiec** [comment]
Hi @sebilasse! Thanks for your idea, but... Hmm, I don't think this kind of functionality should be built in the mongo seeding library.

If you found some library for JSON schema validation, then you can use it in your data definition files using simple import, and after that validate the models before `module.exports`. I feel like this is the proper way to do it, not via including another dependency and figuring out how to provide the JSON schema.

But of course I'm open for a discussion 🙂

**Comment by pkosiec** [comment]
@sebilasse What do you think? Is this solution good enough for you? Or maybe there are some reasons to put the JSON schema validation package into `mongo-seeding`?

BTW I would love to accept a PR with examples of using JSON schema validation support in data import files.
